nucypher/blockchain/eth/actors.py

```diff
@@ -300,10 +283,11 @@ class RitualError(BaseActor.ActorError):
 
     def __init__(
         self,
-        provider_uri: str,  # this is a blockchain connection to the chain with the coordinator contract
+        coordinator_provider_uri: str,
         network: str,  # this must be the network where the coordinator lives
         crypto_power: CryptoPower,
         transacting_power: TransactingPower,
+        condition_provider_uris: Optional[Dict[int, List[str]]] = None,
```

Comment:
Can you help me understand your thinking here?

It seems to me that this might plausibly be a frequent occurrence - is there something special that we need to consider about it?

Reply:
I assume it's ok, but I'm just confirming that if we have two separate URIs for the same chain (one for coordinator, and another for conditions), which is perfectly fine, which means that we create separate connections to the same chain, that we are still good. If we are, then all good.

Related to my question below:
> What if the payment_method and the condition_provider_uris used/had different endpoint URIs for the same chain?

Reply:
I'm not sure how these connections are handled under the hood by web3.  

I surmise it won't be an issue, though.

Since we index by chainID, we uhh... most likely won't have different uris being passed anyway, except when it's overridden?

Reply:
No issue with using the same endpoint for both, and indeed they are deliberately separable.  This decoupling may be desirable if nodes will use a separate endpoint for transacting (dkg participation) vs. read-only (condition evaluation), particularly if the node is configured to use public RPC endpoints for conditions.